src/foam/nanos/auth/PropertyPermissionDAO.js

```diff
@@ -0,0 +1,133 @@
+foam.CLASS({
+    package: 'foam.nanos.auth',
+    name: 'PropertyPermissionDAO',
+  
+    javaImports: [
+      'foam.nanos.auth.AuthService',
+      'foam.core.FObject',
+      'foam.core.PropertyInfo',
+      'foam.core.X',
+      'foam.dao.ProxySink',
+      'java.util.Iterator',
+      'java.util.List'
+    ],
+  
+    extends: 'foam.dao.ProxyDAO',
+    documentation: `A DAO decorator that prevents user from updating / reading
+        properties for which the user does not have the update / read permission`,
+
+    methods: [
+      {
+        name: 'put_',
+        javaCode: `
+    FObject oldObj = getDelegate().find(obj.getProperty("id"));
+    return super.put_(x, hideProperties(x, obj, oldObj, true));
+        `,
+      },
+  
+      {
+        name: 'find_',
+        javaCode: `
+    FObject oldObj = getDelegate().find(id);
```

Comment:
@kgrgreer Should this be `getDelegate().find_(x, id)` since we're the implementer of the method here?